src/overlays/actors/ovl_Demo_Moonend/z_demo_moonend.c

```diff
@@ -28,28 +33,187 @@ ActorInit Demo_Moonend_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Demo_Moonend/func_80C17C48.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Demo_Moonend/DemoMoonend_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Demo_Moonend/func_80C17E70.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Demo_Moonend/func_80C17EE0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Demo_Moonend/DemoMoonend_Draw.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Demo_Moonend/func_80C17FCC.s")
+void DemoMoonend_Init(Actor* thisx, PlayState* play) {
+    DemoMoonend* this = THIS;
+
+    Actor_SetScale(&this->actor, 0.1f);
+    this->actionFunc = DemoMoonend_Idle;
+
+    if (DEMOMOONEND_GET_PARAMS_F(thisx) == 1) {
+        Actor_SetScale(&this->actor, 0.05f);
+        Actor_SetScale(&this->actor, 6.0f);
+        this->actor.draw = NULL;
+        this->cueType = CS_CMD_ACTOR_CUE_558;
+        this->actionFunc = func_80C17B60;
+    } else {
+        Actor_SetScale(&this->actor, 0.095f);
+        func_80183430(&this->skeletonInfo, &object_moonend_Blob_00B5A0, &object_moonend_Blob_001214, this->jointTable,
+                      this->morphTable, NULL);
+        func_801834A8(&this->skeletonInfo, &object_moonend_Blob_001214);
+        this->cueType = CS_CMD_ACTOR_CUE_560;
+        this->actionFunc = func_80C17C48;
+        this->actor.home.rot.z = 0;
+        this->actor.draw = NULL;
+        this->skeletonInfo.frameCtrl.unk_C = 2.0f / 3.0f;
+    }
+}
+
+void DemoMoonend_Destroy(Actor* thisx, PlayState* play) {
+    DemoMoonend* this = THIS;
+
+    if (DEMOMOONEND_GET_PARAMS_F(thisx) != 1) {
+        func_8018349C(&this->skeletonInfo);
+    }
+}
+
+void DemoMoonend_Idle(DemoMoonend* this, PlayState* play) {
+}
+
+void func_80C17B60(DemoMoonend* this, PlayState* play) {
+    u16 cueId;
+
+    if (Cutscene_IsCueInChannel(play, this->cueType)) {
+        cueId = play->csCtx.actorCues[Cutscene_GetCueChannel(play, this->cueType)]->id;
+        Cutscene_ActorTranslateAndYaw(&this->actor, play, Cutscene_GetCueChannel(play, this->cueType));
+        if (this->cueId != cueId) {
+            this->cueId = cueId;
+            switch (this->cueId) {
+                case 1:
+                    this->actor.draw = NULL;
+                    break;
+
+                case 2:
+                    this->actor.draw = func_80C17FCC;
+                    break;
+
+                default:
+                    break;
+            }
+        }
+        if (this->cueId == 2) {
+            func_800B9010(&this->actor, NA_SE_EV_RAINBOW - SFX_FLAG);
+        }
+    } else {
+        this->actor.draw = NULL;
+    }
+}
+
+void func_80C17C48(DemoMoonend* this, PlayState* play) {
+    u16 cueId;
+
+    if (func_80183DE0(&this->skeletonInfo)) {
+        this->actor.home.rot.z = 0;
+    }
+    if (Cutscene_IsCueInChannel(play, this->cueType)) {
+        cueId = play->csCtx.actorCues[Cutscene_GetCueChannel(play, this->cueType)]->id;
```

Comment:
```suggestion
    if (func_80183DE0(&this->skeletonInfo)) {
        this->actor.home.rot.z = 0;
    }
    if (Cutscene_IsCueInChannel(play, this->cueType)) {
        u16 cueId = play->csCtx.actorCues[Cutscene_GetCueChannel(play, this->cueType)]->id;
```
Does moving the define to in-scope work here?

Reply:
I checked and it does match, so might as well.